src/Adapter/MySQL.php

```diff
@@ -336,6 +337,37 @@ private function computeOrderByField(array $filterToTableMapping)
             $orderField = 'p.' . $orderField;
         }
 
+        // put some products at the end of the list
+        $orderField = $this->computeShowLast($orderField);
+
+        return $orderField;
+    }
+
+    /**
+     * Sort product list: InStock, OOPS with qty 0, OutOfStock
+     * @param  string $orderField
+     * @return string
+     */
+    private function computeShowLast($orderField)
+    {
+        // allow only if feature is enabled
+        if(! \Configuration::get('PS_LAYERED_FILTER_SHOW_OUT_OF_STOCK_LAST')) {
+            return $orderField;
+        }
+
+        if ($this->getInitialPopulation() !== null && !empty($orderField)) {
+            $this->getInitialPopulation()->addSelectField('out_of_stock');
+
+            // order by out-of-stock last
+            $byOutOfStockLast = 'p.quantity <= 0';
```

Comment:
I think aliases must be used depending on `$joinMapping['tableAlias']`

Reply:
Done as suggested. Commit: https://github.com/PrestaShop/ps_facetedsearch/pull/154/commits/a5e9acc33eb8aae24174e49f4d68b97e08609005